### PR TITLE
fix: run docker build only on PRs and build+publish only on main+tags

### PR DESCRIPTION
**Pull request by sabineschaller**
## Changes proposed in this pull request

instead of building twice

## Checklist

- [ ] Related issues linked using `fixes #number`
- [ ] Tests added/updated
- [ ] Documentation added
- [ ] Make sure that all checks pass

### CODE REVIEW

File: .github/workflows/docker.yml
Comment (mkurapov) [inline .github/workflows/docker.yml:10]: I just realized why my branches never ended up running the docker release action: I have always used `/` in my branch names:
```yaml
on:
  push:
    branches:
    - '*' // this excludes branches with '/' in them
```

Can you update that to ** please?

Comment (sabineschaller) [reply .github/workflows/docker.yml:10]: Sure! I also want to wait for my other merge to main to actually finish building before I merge this.

**Review by mkurapov — APPROVED** [review]

**Pull request merged**
